File: BackEnd - ControleDeVacinacaoBovina/Models/Venda.cs
using System;

namespace ControleDeVacinacaoBovina.Models
{
    public class Venda
    {
        public int IdVenda { get; set; }
        public int Quantidade { get; set; }
        public int IdOrigem { get; set; }
        public int IdDestino { get; set; }
        public int IdRebanho { get; set; }
        public int IdFinalidadeDeVenda { get; set; }
        public DateTime DataDeVenda { get; set; }
        public bool Ativo { get; private set; } = true;
        public Propriedade Origem { get; set; }
        public Propriedade Destino { get; set; }
        public Rebanho Rebanho { get; set; }
        public FinalidadeDeVenda FinalidadeDeVenda { get; set; }

        public void SetAtivo(bool ativo)
        {
            Ativo = ativo;
        }

    }
}

File: FrontEnd - ControleDeVacinacaoBovina/angular-controle-de-vacinacao-bovina/src/app/Entities/venda.ts
import { FinalidadeDeVenda } from './finalidadedevenda';
import { Especie } from './especie';
import { Propriedade } from './propriedade';
export interface Venda{
   IdVenda: number,
   Quantidade: number,
   IdOrigem: number,
   IdDestino: number,
   IdEspecie: number,
   IdFinalidadeDeVenda: number,
   Ativo: Boolean,
   Origem: Propriedade,
   Destino: Propriedade,
   Especie: Especie,
   FinalidadeDeVenda: FinalidadeDeVenda
}
